src/clj_kondo/impl/analysis/java.clj

```diff
@@ -19,18 +27,40 @@
       (str/replace ".class" "")
       (str/replace ".java" "")))
 
-(defn class->class-name [class-file]
-  (let [bytes (file->bytes class-file)
-        ;; we use ASM for reading the fully qualified class name hand-made
-        ;; solutions, if we ever want to get rid of ASM:
-        ;; https://stackoverflow.com/questions/1649674/resolve-class-name-from-bytecode/1650442#comment115293993_1650442
-        ;; https://stackoverflow.com/a/52332101/6264 Also see working example in
-        ;; Clojure
-        ;; https://gist.github.com/borkdude/d02dc3ff1d03d09351e768964983a46b
-        rdr (new ClassReader ^bytes bytes)
-        class-name (.getClassName rdr)
-        class-name (str/replace class-name "/" ".")]
-    class-name))
+(def ^:private opcode->flags
+  {Opcodes/ACC_PUBLIC #{:public}
+   Opcodes/ALOAD #{:public :field :static}
+   Opcodes/SIPUSH #{:public :field :final}
+   Opcodes/LCONST_0 #{:public :method :static}})
+
+(defn ^:private class-is->class-info
+  "Parse class-bytes using ASM."
+  [^InputStream class-is]
+  (let [class-reader (ClassReader. (input-stream->bytes class-is))
```

Comment:
We may need `with-open` here too? Not sure.

Reply:
I don't think we need here because that private function already does that [here](https://github.com/clj-kondo/clj-kondo/pull/2008/files#diff-40ca05e82ee1bafb0e01402be6bc6119871dc41c1b65f9b830634d94f3ac1b10R18), only returning the byte array